tests/v4/test_basegrid.py

```diff
@@ -35,7 +35,7 @@ def get_axis_dim(self, axis: str) -> int:
 def test_basegrid_ravel_unravel_index(grid):
     axes = grid.axes
     dimensionalities = (grid.get_axis_dim(axis) for axis in axes)
-    all_possible_axis_indices = itertools.product(*[range(dim) for dim in dimensionalities])
+    all_possible_axis_indices = itertools.product(*[np.arange(dim)[:, np.newaxis] for dim in dimensionalities])
```

Comment:
Added 9826c9ae644c4541ca18d2dc9c1f63c51a2f189e (since many of the original changes weren't required)

Reply:
Hmm, I intentionally made these changes because the `position` dictionary now has lists of indices, rather than individual integers. 

So I very intentionally made the change to test_basegrid.py; to check whether an input like `TestGrid({"Z": 10, "Y": 20, "X": 30})` would work. I'd prefer to keep testing that particular set-up therefore